tests/io/test_hive_style_partitions.py

```diff
@@ -0,0 +1,76 @@
+import pyarrow as pa
+import pytest
+
+import daft
+
+
+@pytest.fixture(scope="session")
+def public_storage_io_config() -> daft.io.IOConfig:
+    return daft.io.IOConfig(
+        azure=daft.io.AzureConfig(storage_account="dafttestdata", anonymous=True),
+        s3=daft.io.S3Config(region_name="us-west-2", anonymous=True),
+        gcs=daft.io.GCSConfig(anonymous=True),
+    )
+
+
+def check_file(public_storage_io_config, read_fn, uri):
```

Comment:
Couple suggestions for more robust test coverage.
- Write partitioned tables to a temp dir via `pyarrow write dataset` api, set the partitioning to the partition columns and partitioning_flavor = 'HIVE', read_back using daft and pyarrow and assert equal.
- Write partitioned tables to a temp dir via daft, read_back using daft and assert equal.
- Test the most common datatypes for partitioning, strings, dates, timestamps, ints. Test for NULLs as well (nulls are __HIVE_DEFAULT_PARTITION__).
- Try to separate out tests and reduce the number of assertions per test, especially if the assertions test different functionalities . i.e. you can have a test that reads, and another test that reads with pushdowns. This may be a personal preference, but I do believe smaller tests are easier to reason about and debug.

Reply:
Thanks for the suggestions! Updated the tests accordingly.

Had to do some wonky stuff due to internal inconsistencies with handling timestamps, and null handling with CSV.